pinot-core/src/main/java/com/linkedin/pinot/core/io/reader/impl/v1/BaseChunkSingleValueReader.java

```diff
@@ -66,18 +70,49 @@ public BaseChunkSingleValueReader(PinotDataBuffer pinotDataBuffer, ChunkDecompre
 
     _lengthOfLongestEntry = _dataBuffer.getInt(headerOffset);
     headerOffset += INT_SIZE;
+
+    int dataHeaderStart = headerOffset;
+    if (version > 1) {
+      _totalDocs = _dataBuffer.getInt(headerOffset);
+      headerOffset += INT_SIZE;
+
+      ChunkCompressorFactory.CompressionType compressionType =
+          ChunkCompressorFactory.CompressionType.values()[_dataBuffer.getInt(headerOffset)];
+      _chunkDecompressor = ChunkCompressorFactory.getDecompressor(compressionType);
+      headerOffset += INT_SIZE;
+
+      dataHeaderStart = _dataBuffer.getInt(headerOffset);
+    } else {
+      _totalDocs = -1; // No way to deduce total docs in v1 format.
+      _chunkDecompressor = ChunkCompressorFactory.getDecompressor(ChunkCompressorFactory.CompressionType.SNAPPY);
+    }
+
     _chunkSize = (_lengthOfLongestEntry * _numDocsPerChunk);
 
     // Slice out the header from the data buffer.
-    int headerLength = _numChunks * INT_SIZE;
-    _header = _dataBuffer.view(headerOffset, headerOffset + headerLength);
+    int dataHeaderLength = _numChunks * INT_SIZE;
+    int rawDataStart = dataHeaderStart + dataHeaderLength;
+    _dataHeader = _dataBuffer.view(dataHeaderStart, rawDataStart);
+
+    // Useful for uncompressed data.
+    _rawData = _dataBuffer.view(rawDataStart, _dataBuffer.size());
   }
 
   @Override
   public void close() {
     // Nothing to close here.
   }
 
+  /**
+   * Returns the total number of docs in the file. This is only support from v2 onwards.
+   * Returns -1 for v1.
+   *
+   * @return Total docs for v2 onwards, -1 for v1.
+   */
+  public int getTotalDocs() {
```

Comment:
why do we need this method.

Reply:
It was needed in intermediate state of the code, but ended up being unused. However, IDE did not show it as unused, so missed to clean it up. Will clean.